### PR TITLE
PRIME-2764 Update validation to require only one email address for share GPID page

### PR DESCRIPTION
**Pull request by bergomi02**
- Update validation to require only one email address on the share GPID page
- change email format validation from default email validation, instead of custom one

### CODE REVIEW

**Comment by neophyte57** [comment]
Not sure if this PR is ready or not but getting this situation which doesn't seem correct 🤔
image

**Comment by bergomi02** [comment]
> Not sure if this PR is ready or not but getting this situation which doesn't seem correct 🤔 image

it should be fixed now.

**Comment by neophyte57** [comment]
Getting this error now:

image

**Comment by bergomi02** [comment]
> Getting this error now:
> 
> image

Good catch. It is fixed.

File: prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts [hunk 1]
```diff
@@ -189,37 +190,51 @@ export class PharmanetEnrolmentSummaryComponent extends BaseEnrolmentPage implem
...
-      )
-      .subscribe(() => {
-        let emails = new Array<string>();
-        this.careSettingConfigs.forEach((config) => {
-          emails.push(config.formArray.value.map(email => email.email));
+
+    if (!this.atLeastOneEmailFilled()) {
+      const data: DialogOptions = {
+        title: 'Missing Email',
+        message: `Please enter at least one email for the Approval Notification?`,
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts:197]: I think question mark at end of sentence is unnecessary and icon should be "warning" or "error"

File: prime-angular-frontend/src/app/modules/enrolment/pages/next-steps/next-steps.component.ts [hunk 2]
```diff
@@ -171,38 +172,51 @@ export class NextStepsComponent extends BaseEnrolmentProfilePage implements OnIn
...
-      .subscribe(() => {
-        this.toastService.openSuccessToast('Email was successfully sent');
-        this.router.navigate([EnrolmentRoutes.PHARMANET_ENROLMENT_SUMMARY],
-          { relativeTo: this.route.parent, queryParams: { initialEnrolment: this.initialEnrolment } });
-      });
-    this.onPageChange({ atEnd: true });
+    if (!this.atLeastOneEmailFilled()) {
+      const data: DialogOptions = {
+        title: 'Missing Email',
+        message: `Please enter at least one email for the Approval Notification?`,
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/enrolment/pages/next-steps/next-steps.component.ts:179]: I think question mark at end of sentence is unnecessary and icon should be "warning" or "error"

**Comment by neophyte57** [comment]
Need to re-address issue of leading/trailing spaces, in this implementation 
image

File: prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts (hunk 1 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts]: Update as suggested.

File: prime-angular-frontend/src/app/modules/enrolment/pages/next-steps/next-steps.component.ts (hunk 2 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/enrolment/pages/next-steps/next-steps.component.ts]: Update as suggested.

**Comment by bergomi02** [comment]
> Need to re-address issue of leading/trailing spaces, in this implementation image

Interesting found. Space is not allowed now. It should fix the issue.

File: prime-angular-frontend/src/app/shared/components/forms/email-form/email-form.component.html [hunk 3]
```diff
@@ -8,6 +8,7 @@
           <input matInput
                  placeholder="Email"
                  formControlName="email"
+                 (keydown.space)="$event.preventDefault()"
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/forms/email-form/email-form.component.html:11]: This solution doesn't work for emails copied and pasted from somewhere else  😒

Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/forms/email-form/email-form.component.html]: Created a directive to trim the space. It should work better.

File: prime-angular-frontend/src/app/shared/shared.module.ts [hunk 4]
```diff
@@ -130,6 +130,7 @@ import { HealthAuthSiteOverviewContainerComponent } from './components/health-au
 import { EmailFormComponent } from './components/forms/email-form/email-form.component';
 import { ChangeVendorNoteComponent } from './components/dialogs/content/change-vendor-note/change-vendor-note.component';
 import { TransferHASiteComponent } from './components/dialogs/content/transfer-ha-site/transfer-ha-site.component';
+import { TrimSpaceInputDirective } from '@lib/modules/forms/trim-space.directive';
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/shared.module.ts:133]: Better to group with `UpperCaseInputDirective` further up

Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/shared.module.ts]: Updated as suggested.

File: prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts [hunk 5]
```diff
@@ -71,6 +71,7 @@ export class PharmanetEnrolmentSummaryComponent extends BaseEnrolmentPage implem
     private dialog: MatDialog,
     private toastService: ToastService,
     private configService: ConfigService,
+    private formUtilsService: FormUtilsService,
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts:74]: Unused?

**Review by neophyte57 — APPROVED** [review]
Nicely done! 👍

File: prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts (hunk 5 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.ts]: removed.

**Pull request merged**